src/rules/no_unused_vars.rs

```diff
@@ -0,0 +1,2052 @@
+  fn no_unused_vars_ts_ok_9() {
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+function foo(): Nullable {
+  return null;
+}
+foo();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+import { SomeOther } from 'some';
+import { Another } from 'some';
+class A extends Nullable<SomeOther> {
+  do(a: Nullable<Another>) {
+    console.log(a);
+  }
+}
+new A();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+import { SomeOther } from 'some';
+import { Another } from 'some';
+interface A extends Nullable<SomeOther> {
+  other: Nullable<Another>;
+}
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+import { SomeOther } from 'some';
+import { Another } from 'some';
+interface A extends Nullable<SomeOther> {
+  do(a: Nullable<Another>);
+}
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Foo } from './types';
+class Bar<T extends Foo> {}
+new Bar<number>();
+      ",
+    );
+  }
+
+  #[test]
+  fn no_unused_vars_ts_ok_10() {
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Foo, Bar } from './types';
+class Baz<T extends Foo & Bar> {}
+new Baz<any>();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Foo } from './types';
+class Bar<T = Foo> {}
+new Bar<number>();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Foo } from './types';
+class Foo<T = any> {}
+new Foo();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Foo } from './types';
+class Foo<T = {}> {}
+new Foo();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Foo } from './types';
+class Foo<T extends {} = {}> {}
+new Foo();
+      ",
+    );
+  }
+
+  #[test]
+  fn no_unused_vars_ts_ok_11() {
+    assert_lint_ok::<NoUnusedVars>(
+      "
+type Foo = 'a' | 'b' | 'c';
+type Bar = number;
+export const map: { [name in Foo]: Bar } = {
+  a: 1,
+  b: 2,
+  c: 3,
+};
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+class A<T> {
+  bar: T;
+}
+new A<Nullable>();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+import { SomeOther } from 'other';
+function foo<T extends Nullable>() {}
+foo<SomeOther>();
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+import { SomeOther } from 'other';
+class A<T extends Nullable> {
+  bar: T;
+}
+new A<SomeOther>()
+      ",
+    );
+
+    assert_lint_ok::<NoUnusedVars>(
+      "
+import { Nullable } from 'nullable';
+import { SomeOther } from 'other';
+interface A<T extends Nullable> {
+  bar: T;
+}
+export const a: A<SomeOther> = {
+  foo: 'bar',
+};
+
+      ",
+    );
+  }
+
+  #[test]
+  #[ignore = "typescript property analysis is not implemented yet"]
```

Comment:
Is this ignore still relevant?

Reply:
Yes, it's not implemented yet. It requires more delicate scope analysis I think